porch/pkg/cache/repository.go

```diff
@@ -68,6 +72,20 @@ func (c *cachedPackageRevision) GetPackageRevision() *v1alpha1.PackageRevision {
 
 var _ repository.PackageRevision = &cachedPackageRevision{}
 
+type cachedPackage struct {
+	repository.Package
+	latestPackageRevision string
```

Comment:
In the current PR, deletion flushes the cache. I didn't consider approval, but I will update the PR so that approval flushes the latestPackageRevision part of the cache.

Reply:
So it looks like with packagerevisions, the latest package revisions gets updated in the cached package revisions every time a draft "closes" (which includes approval. I added a TODO this PR (in `(r *cachedRepository) update`) to also update the cached packages' latest revisions in the same place, which I can implement in the next PR.